control-box/ASRFader.h

```diff
@@ -0,0 +1,153 @@
+// Statuses for the PIR sensor
+#define PIR_OFF 0
+#define PIR_FADEIN 1
+#define PIR_ON 2
+#define PIR_FADEOUT 3
+
+/**
+ * A fader with fade-in, sustain, and fade-out. Extended from the Light Hoops' PIRSensor.
+ * 
+ * This works well with a PIR motion sensor. Set it for minimum cycle time, so they have
+ * a period of about 1-2 seconds. Use this to sustain patterns for the appropriate duration.
+ *
+ * With the fade in & out, you can build smooth transitions between the "on" and "off" states.
+ * 
+ * @author Sam Minnee
+ */
+class ASRFader {
```

Comment:
So this is teh state machine that keeps track of the PIR status and keeps track of the fade value?